exaudfclient/base/script_options_parser/ctpg/script_option_lines_ctpg.h

```diff
@@ -0,0 +1,54 @@
+#ifndef SCRIPTOPTIONLINESCTPG_H
+#define SCRIPTOPTIONLINESCTPG_H
+
+#include <string>
+#include <functional>
+#include <vector>
+#include <map>
+#include <ostream>
+
+namespace ExecutionGraph
+{
+
+namespace OptionsLineParser
+{
+
+namespace CTPG
+{
+
+class ParserResult;
+
+struct ScriptOption {
+    std::string value;
+    size_t idx_in_source;
+    size_t size;
+
+    bool operator==(const ScriptOption & right) const {
+        return value == right.value && idx_in_source == right.idx_in_source && size == right.size;
+    }
+    friend void PrintTo(const ScriptOption& option, std::ostream* os) {
```

Comment:
why friend, don't see the reason directly, or is it, because you can use PrintTo(option, stream), so it is not a method

Reply:
Copied the example from https://github.com/google/googletest/blob/main/docs/advanced.md#teaching-googletest-how-to-print-your-values

Reply:
Does it make sense to add the link as comment to the code

Reply:
Done